test/conformance/service_test.go

```diff
@@ -536,4 +536,47 @@ func TestReleaseService(t *testing.T) {
 	}
 }
 
+func TestMultipleNamespace(t *testing.T) {
+	t.Parallel()
+
+	altServiceNamespace := fmt.Sprintf("%s-%s", test.ServingNamespace, test.ObjectNameForTest(t))
+	serviceName := test.ObjectNameForTest(t)
+
+	defaultClients := e2e.Setup(t) // This one uses the default namespace `test.ServingNamespace`
+	altClients := e2e.SetupWithNamespace(t, altServiceNamespace)
+
+	_, err := test.CreateNamespace(altClients, altServiceNamespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.DeleteNamespace(altClients, altServiceNamespace)
```

Comment:
I suspect that CreateNamespace could return an error despite successfully creating the namespace such a timeout on http response. Given that you don't need to validate creation to perform a delete operation why not move defer up further?